### PR TITLE
Allow offset compensation for zero point via probing macros

### PR DESCRIPTION
**Pull request by JustinSGray**

### CODE REVIEW

File: PROBEYSLOT [hunk 1]
```diff
@@ -26,7 +27,7 @@ G65 "PROBECONFIG"
 
 (CALCULATE EXTENDED WCS NUMBER)
 (FIX is a round down function and MOD is modulo)
-#114 = MOD[#1,FIX[#1]]*10 
+#114 = [[#1 - FIX[#1]] * 10]
```
Comment (JustinSGray) [inline PROBEYSLOT:30]: good catch!

File: syil_lnc_toolpath.cps [hunk 2]
```diff
@@ -1673,23 +1675,49 @@ function onCyclePoint(x, y, z) {
       writeBlock(gFormat.format(65), '"PROTECTEDMOVE"', zOutput.format(z - cycle.depth));
 
       WCS_CODE = getProbingArguments(cycle, probeWorkOffsetCode);
+    
+      if(cycle.approach1== "positive")
```
Comment (JustinSGray) [inline syil_lnc_toolpath.cps:1679]: I think you could replace the `if` statement here with the `approach` function. That converts it to a sign, which you could use to have only a single set of equations.

File: syil_lnc_toolpath.cps [hunk 3]
```diff
@@ -1673,23 +1675,49 @@ function onCyclePoint(x, y, z) {
...
+      writeBlock(gFormat.format(65), '"PROBEX"', WCS_CODE[7], WCS_CODE[8], B_ARG);
+      writeBlock(gFormat.format(65), '"CHECKPOSITIONALTOLERANCE"', WCS_CODE[8], WCS_CODE[9], WCS_CODE[3],'V1', EXPECTED_X, EXPECTED_Y, EXPECTED_Z);
       break;
     case "probing-y":
       forceXYZ();
       // writeBlock(gFormat.format(31), "P2 ", zOutput.format(z - cycle.depth), "F50");  // protected positioning move 
       writeBlock(gFormat.format(65), '"PROTECTEDMOVE"', zOutput.format(z - cycle.depth));
 
       WCS_CODE = getProbingArguments(cycle, probeWorkOffsetCode);
+      if(cycle.approach1== "positive")
```
Comment (JustinSGray) [inline syil_lnc_toolpath.cps:1704]: check out the `approach` function. That can be used to simplify this code a bit.

**Review by JustinSGray — COMMENTED** [review]
I had only very minor changes suggested to the post. You don't need the if statements around the `argument` stuff. There is already a function that convert it to a simple sign you can use in the equations.

**Comment by Mrsmith1231** [comment]
I updated the post to use the approach function.

**Pull request merged**